File: converters/src/main/scala/hr/element/pgscala/converters/PGNullableUUIDConverter.scala
package hr.element.pgscala.converters

import java.util.UUID

object PGNullableUUIDConverter extends PGTypeConverter[Option[UUID]] {
  def toPGString(value: Option[UUID]): String =
    value match {
      case Some(u) =>
        PGUUIDConverter.toPGString(u)
      case _ =>
        null
    }

  def fromPGString(value: String): Option[UUID] =
    if(value != null && value.nonEmpty) {
      Some(PGUUIDConverter.fromPGString(value))
    }
    else {
      None
    }
}

File: pgjava-converters/src/main/java/hr/element/pgscala/converters/PGNullableUUIDConverter.java
package hr.element.pgscala.converters;

import org.joda.convert.*;

import java.util.UUID;

public enum PGNullableUUIDConverter implements StringConverter<UUID> {
  INSTANCE;

  public static final String pgType = "text";

  @ToString
  public static String uUIDToString(final UUID uuid) {
    return null == uuid ? null : uuid.toString();
  }

  @FromString
  public static UUID stringToUUID(final String uuid) {
    return null == uuid ? null : UUID.fromString(uuid);
  }

// ----------------------------------------------------------------------------

  public String convertToString(final UUID uuid) {
    return uUIDToString(uuid);
  }

  public UUID convertFromString(final Class<? extends UUID> clazz, final String uuid) {
    return stringToUUID(uuid);
  }
}
